lib/Common/Codex/Utf8Helper.h

```diff
@@ -72,6 +77,7 @@ namespace utf8
         }
         else
         {
+            AnalysisAssert(sourceCount * 3 + 1 <= destCount);
```

Comment:
Just to check, this means that each pair of characters in the original string encodes to a utf8char of at most 3 bytes long, plus the null terminator? There's no cases where we have strings of surrogate pairs that each translate to a 4-byte utf8 encoding?

Reply:
This assert isn't correct (as the assertion failures point out). This would be a safe over-estimation, but in a number of cases (especially when using this `NoAlloc` version) we call this function first with `destString == nullptr` to get a precise count of how much space is needed, and then call in again with a correctly sized buffer, which will in general be smaller than `sourceCount * 3 + 1`

Reply:
Since the assert is only enabled in debug builds (or SAL runs) you could maybe `AnalysisAssert(destCount >= utf8::CountTrueUtf8(sourceString, cchSourceString))`? Not sure if SAL is sophisticated enough to get much out of it though.